code/game/supplyshuttle.dm

```diff
@@ -447,14 +521,48 @@ var/datum/controller/supply/supply_controller = new()
 		qdel(C)
 
 	// Sell manifests.
-	for(var/atom/movable/MA in area_shuttle)
-		if(istype(MA, /obj/item/paper/manifest))
-			var/obj/item/paper/manifest/M = MA
+	for(var/atom/movable/movable_atom in area_shuttle)
+		if(istype(movable_atom, /obj/item/paper/manifest))
+			var/obj/item/paper/manifest/M = movable_atom
 			if(M.stamped && M.stamped.len)
 				points += points_per_slip
 
+		//black market points
+		if(black_market_enabled)
+			var/points_to_add = get_black_market_value(movable_atom)
+			if(points_to_add == KILL_MENDOZA)
+				kill_mendoza()
+			black_market_sold_items[movable_atom.type] += 1
+			black_market_points += points_to_add
+
+		// Don't disintegrate humans! Maul their corpse instead. >:)
+		if(ishuman(movable_atom))
+			var/timer = 0.5 SECONDS
+			for(var/index in 1 to 10)
+				timer += 0.5 SECONDS
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(maul_human), movable_atom), timer)
+
 		// Delete everything else.
-		qdel(MA)
+		else qdel(movable_atom)
+
+/proc/maul_human(var/mob/living/carbon/human/mauled_human)
```

Comment:
can't you scope this to be an ominous proc on the supply controller instead of a global proc that will always fuck with the supply controller

Reply:
it's also used on normally falling onto the pit but sure